omx/gstomxvideofilter.c

```diff
@@ -0,0 +1,2575 @@
+    if (port_def.format.video.eColorFormat == OMX_COLOR_FormatUnused) {
+      GST_ERROR_OBJECT (self, "Unsupported format %s",
+          gst_video_format_to_string (outinfo->finfo->format));
+      return FALSE;
+    }
+    port_def.nBufferAlignment = 0;
+    port_def.bBuffersContiguous = 0;
+    port_def.nBufferCountActual = self->output_buffers;
+
+    /* scalar buffer pitch should be multiple of 16 */
+    switch (port_def.format.video.eColorFormat) {
+      case OMX_COLOR_FormatYUV420SemiPlanar:
+        port_def.format.video.nStride =
+            ((port_def.format.video.nFrameWidth + 15) & 0xfffffff0);
+        port_def.nBufferSize =
+            port_def.format.video.nStride * port_def.format.video.nFrameHeight *
+            3 / 2;
+        break;
+      case OMX_COLOR_FormatYCbYCr:
+        port_def.format.video.nStride =
+            ((port_def.format.video.nFrameWidth + 15) & 0xfffffff0) * 2;
+        port_def.nBufferSize =
+            port_def.format.video.nStride * port_def.format.video.nFrameHeight;
+        break;
+      default:
+        GST_ERROR_OBJECT (self,
+            "Component supports unsupported color format %d",
+            port_def.format.video.eColorFormat);
+        break;
+    }
+
+    GST_DEBUG_OBJECT (self, "Updating outport port definition");
+    if (gst_omx_port_update_port_definition (outport->data,
+            &port_def) != OMX_ErrorNone)
+      return FALSE;
+
+    /* Configure output pool */
+    config = gst_buffer_pool_get_config (outpool->data);
+    gst_buffer_pool_config_set_params (config, srccaps->data,
+        port_def.nBufferSize, port_def.nBufferCountActual,
+        port_def.nBufferCountActual);
+    if (!gst_buffer_pool_set_config (outpool->data, config))
+      goto config_pool_failed;
+
+    GST_LOG_OBJECT (self,
+        "Port defs: nFrameWidth: %d, nFrameHeight: %d, CompressionFormat: %d, ColorFormat: %d, BufferAlignment: %d, BufferContiguous: %d, BufferCountActual: %d, VideoStride: %d, Buffersize: %d",
+        port_def.format.video.nFrameWidth,
+        port_def.format.video.nFrameHeight,
+        port_def.format.video.eCompressionFormat,
+        port_def.format.video.eColorFormat, port_def.nBufferAlignment,
+        port_def.bBuffersContiguous, port_def.nBufferCountActual,
+        port_def.format.video.nStride, port_def.nBufferSize);
+
+    srcinfo = srcinfo->next;
+    srccaps = srccaps->next;
+    outpool = outpool->next;
+  }
+
+  if (klass->set_format) {
+    if (!klass->set_format (self, incaps, ininfo, outcaps_list, outinfo_list)) {
+      GST_ERROR_OBJECT (self, "Subclass failed to set the new format");
+      return FALSE;
+    }
+  }
+
+  return TRUE;
+
+drain_failed:
+  {
+    GST_ERROR_OBJECT (self, "Failed to reconfigure, cannot drain component");
+    return FALSE;
+  }
+config_pool_failed:
+  {
+    GST_INFO_OBJECT (self, "Failed to set config on output pool");
+    return FALSE;
+  }
+}
+
+/* given a fixed @caps on @pad, create the best possible caps for the
```

Comment:
Some functions have big comments like this one, other functions such as _shutdown, _close, etc .. don't. Standarize ?

Reply:
We should add doxygen documentation to all functions in all class files into this project.

Reply:
In general the documentation follows the gstreamer style.